src/optimagic/visualization/history_plots.py

```diff
@@ -8,36 +8,51 @@
 import plotly.graph_objects as go
 from pybaum import leaf_names, tree_flatten, tree_just_flatten, tree_unflatten
 
-from optimagic.config import PLOTLY_PALETTE, PLOTLY_TEMPLATE
+from optimagic.config import PLOTLY_TEMPLATE
 from optimagic.logging.logger import LogReader, SQLiteLogOptions
 from optimagic.optimization.algorithm import Algorithm
 from optimagic.optimization.history import History
 from optimagic.optimization.optimize_result import OptimizeResult
 from optimagic.parameters.tree_registry import get_registry
 from optimagic.typing import IterationHistory, PyTree
+from optimagic.visualization.backends import get_plot_backend_class
+from optimagic.visualization.plotting_utilities import LineData, get_palette_cycle
 
-OptimizeResultOrPath = OptimizeResult | str | Path
+BACKEND_TO_CRITERION_PLOT_LEGEND_PROPERTIES: dict[str, dict[str, Any]] = {
+    "plotly": {
+        "yanchor": "top",
+        "xanchor": "right",
+        "y": 0.95,
+        "x": 0.95,
+    },
+    "matplotlib": {
+        "loc": "upper right",
+    },
+}
+
+
+ResultOrPath = OptimizeResult | str | Path
 
 
 def criterion_plot(
-    results: OptimizeResultOrPath
-    | list[OptimizeResultOrPath]
-    | dict[str, OptimizeResultOrPath],
+    results: ResultOrPath | list[ResultOrPath] | dict[str, ResultOrPath],
     names: list[str] | str | None = None,
     max_evaluations: int | None = None,
-    template: str = PLOTLY_TEMPLATE,
-    palette: list[str] | str = PLOTLY_PALETTE,
+    backend: str = "plotly",
```

Comment:
Can you make `backend` a literal with the respective entries? When experimenting I realized that there is not autocomplete for the backend yet.